floss/main.py

```diff
@@ -534,7 +542,7 @@ def main(argv=None) -> int:
     lang_id = identify_language(sample, static_strings)
 
     # set language configurations
-    if lang_id == Language.GO:
+    if (lang_id == Language.GO and args.language == "") or args.language == Language.GO.value:
         results.metadata.language = Language.GO.value
```

Comment:
```suggestion
    if args.language == Language.GO.value or lang_id == Language.GO:
        results.metadata.language = Language.GO.value
```

Reply:
would this work?

Reply:
Considering the possibility of false positives, what if our language identifier incorrectly detects a binary as Go, even though it isn't?

Reply:
then user's have to specify `--language none` explicitly?

the conditions should be similar to your code here, or are they not?

Reply:
> then user's have to specify `--language none` explicitly?

Indeed, if there are false positives, they have the option to do so.

> the conditions should be similar to your code here, or are they not?

I may be wrong, but the code appears to already enable users to specify the language explicitly using the --language flag and the ability to override the automatic detection if necessary?

Reply:
my second comment is regarding the logic of the `if` here, my goal was to simplify it

Reply:
the same then applies to the ones below

Reply:
I wanted to provide a contradictory example:
```
if args.language == Language.GO.value or lang_id == Language.GO:
    results.metadata.language = Language.GO.value
```

Consider a scenario where we have a Rust binary that is incorrectly identified as Go by our identification script. However, the user is aware that it is a Rust binary. In this case, the `lang_id` variable would be `True`. Since there is an "OR" statement in the expression, it will evaluate to `True`.

I believed this example could be relevant in highlighting how the expression may lead to inaccurate language detection under specific circumstances.

I am open to further discussion. Thanks😄

Reply:
good point, let's leave as is